pkg/ext/apiserver.go

```diff
@@ -0,0 +1,259 @@
+package ext
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metainternalversion "k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apiserver/pkg/authentication/authenticator"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/apiserver/pkg/endpoints/openapi"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	genericoptions "k8s.io/apiserver/pkg/server/options"
+	openapicommon "k8s.io/kube-openapi/pkg/common"
+	"k8s.io/kube-openapi/pkg/validation/spec"
+)
+
+var (
+	schemeBuilder = runtime.NewSchemeBuilder(addKnownTypes, metainternalversion.AddToScheme)
+	AddToScheme   = schemeBuilder.AddToScheme
+)
+
+func addKnownTypes(scheme *runtime.Scheme) error {
+	metav1.AddToGroupVersion(scheme, schema.GroupVersion{Version: "v1"})
+	return nil
+}
+
+type ExtensionAPIServerOptions struct {
+	// GetOpenAPIDefinitions is collection of all definitions. Required.
+	GetOpenAPIDefinitions             openapicommon.GetOpenAPIDefinitions
+	OpenAPIDefinitionNameReplacements map[string]string
+
+	// Authenticator will be used to authenticate requests coming to the
+	// extension API server. Required.
+	Authenticator authenticator.Request
+
+	// Authorizer will be used to authorize requests based on the user,
+	// operation and resources. Required.
+	//
+	// Use [NewAccessSetAuthorizer] for an authorizer that uses Steve's access set.
+	Authorizer authorizer.Authorizer
+
+	// Listener is the TCP listener that is used to listen to the extension API server
+	// that is reached by the main kube-apiserver. Required.
+	Listener net.Listener
```

Comment:
The field doesn't do much yet but due to the library being used it's necessary to have. In the next PR it will be actually useful because the HTTPS server is using it.